### PR TITLE
Doc: Data model schema

### LINKED ISSUE #531: Doc: Add schema for cv_image_v3
### Enhancement summary

Add schema for cv_image_v3

### Which component of AVD is impacted

cv_image_v3

### Use case example

N/A

### Describe the solution you'd like

N/A

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by sugetha24**
## Change Summary

Added a new `Schema` section to the web documentation and linked this page from the `module documentation` section 

## Related Issue(s)

Fixes #531 

## Proposed changes
Added a new `Schema documentation` section to the web documentation. This contains all the schema related documentations. This page is linked from the `module documentation` section.

## How to test
run `make webdoc` from main repo folder
`pip3 install -r docs/requirements.txt`
`mkdocs serve`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly. (check the box if not applicable)
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
   - Returns number of created and/or deleted containers
 options:
   devices:
-    description: List of devices with their container and configlets information
+    description: List of devices with their container, configlet and image bundle information
```
Comment (sugetha24) [inline ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py]: One of the previous commits directly modified the .rst.md file without adding them to the docstrings first and running `make webdoc` as that is the right way to do it. I have updated the change in this PR.

File: ansible_collections/arista/cvp/docs/schema/cv_container_v3.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Schema for cv_container_v3
+
+| Variable | Type | Required | Default | Choices | Description |
+| -------- | ---- | -------- | ------- | ------------------ | ----------- |
+| apply_mode | str | No | loose | loose<br>strict | Set how configlets are attached/detached on container. If set to strict all configlets not listed in your vars are detached. |
```
Comment (noredistribution) [inline ansible_collections/arista/cvp/docs/schema/cv_container_v3.md]: nit: `Set how configlets are attached/detached to containers. If set to strict, all configlets not listed in your vars will be detached`

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Schema for cv_device_v3
+
+| Variable | Type | Required | Default | Choices | Description |
+| -------- | ---- | -------- | ------- | ------------------ | ----------- |
+| apply_mode | str | No | loose | loose<br>strict | Set how configlets are attached/detached on device. If set to strict all configlets not listed in your vars are detached |
```
Comment (noredistribution) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md:5]: nit:
`Set how configlets are attached/detached to devices. If set to strict, all configlets and image bundles not listed in your vars will be detached`

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+| search_key | str | No | hostname | fqdn<br>hostname<br>serialNumber | Key name to use to look for device in Cloudvision |
+| state | str| No | present | present<br>factory_reset | Set if ansible should build or remove devices on Cloudvision |
```
Comment (noredistribution) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]: please add the new states too
`provisioning_reset` and `absent`

we should also update the description from `Set if ansible should build or remove devices on CLoudvision` to 
`Set if ansible should build, remove devices from provisioning, fully decommission or factory reset devices on CloudVision`

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md (hunk 3 above)
Comment (noredistribution) [reply ansible_collections/arista/cvp/docs/schema/cv_device_v3.md:5]: we probably need to update the docstring in cv_device_v3.py too?

File: ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+| Variable | Type | Required | Default | Choices | Description |
+| -------- | ---- | -------- | ------- | ------------------ | ----------- |
+| change_id | list | No |  |  | List of change IDs to get/remove |
+| name | str | No |  |  | The name of the change control. If not provided, one will be generated |
+| state | str | No | show | show<br>set<br>remove | Set if we should get, set/update, or remove the change control |
+| change | Dict | No |  |  | A dict containing the change control to be created/modified |
+| &nbsp;&nbsp;&nbsp;&nbsp;name | str | No |  |  | Name of change control |
+| &nbsp;&nbsp;&nbsp;&nbsp;notes | str | No |  |  | Any notes that you want to add |
+| &nbsp;&nbsp;&nbsp;&nbsp;stages | Dict | Yes |  |  |  |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-name | str | Yes |  |  | Name of stage |
```
Comment (noredistribution) [inline ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]: we shold have space in `-name` I think and same for the other one further down

**Review by noredistribution — CHANGES_REQUESTED** [review]

File: mkdocs.yml [hunk 6]
```diff
@@ -137,6 +137,16 @@ nav:
       - Get facts from CloudVision: docs/how-to/v3/cv_facts_v3.md
     - Development tips & tricks: docs/installation/collection-installation.md
     - Activate logging: docs/how-to/debug.md
+  - Schema documentation:
+    - CV Modules v3:
+      # - Module cv_configlet_v3: docs/schema/cv_tag_v3.md
+      - Module cv_container_v3: docs/schema/cv_container_v3.md
```
Comment (JulioPDX) [inline mkdocs.yml]: I would suggest removing the "module" prefixes here, it's already stated in the upper link

File: ansible_collections/arista/cvp/docs/schema/cv_container_v3.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+| state | str | No | present | present<br>absent | Set if ansible should build or remove devices on Cloudvision |
```
Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_container_v3.md]:
```suggestion
| state | str | No | present | present<br>absent | Set if Ansible should build or remove devices on CloudVision |
```

File: ansible_collections/arista/cvp/docs/schema/cv_container_v3.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+| topology | dict | Yes |  |  | Yaml dictionary to describe intended containers |
```
Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_container_v3.md]:
```suggestion
| topology | dict | Yes |  |  | YAML dictionary to describe intended containers |
```

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Schema for cv_device_v3
+
+| Variable | Type | Required | Default | Choices | Description |
+| -------- | ---- | -------- | ------- | ------------------ | ----------- |
+| apply_mode | str | No | loose | loose<br>strict | Set how configlets are attached/detached on device. If set to strict all configlets not listed in your vars are detached |
+| search_key | str | No | hostname | fqdn<br>hostname<br>serialNumber | Key name to use to look for device in Cloudvision |
```
Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]:
```suggestion
| search_key | str | No | hostname | fqdn<br>hostname<br>serialNumber | Key name to use to look for device in CloudVision |
```

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md (hunk 4 above)
Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]:
```suggestion
| state | str| No | present | present<br>factory_reset | Set if Ansible should build or remove devices on CloudVision |
```

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+| state | str| No | present | present<br>factory_reset | Set if ansible should build or remove devices on Cloudvision |
+| devices | List | Yes |  |  | List of devices with their container and configlets information |
+| &nbsp;&nbsp;&nbsp;&nbsp;ipAddress | str | Yes |  |  | IP address of the device |
+| &nbsp;&nbsp;&nbsp;&nbsp;fqdn | str | Yes |  |  | Fully Qualified Domain Name of the device.<br>This field is required along with parentContainerName |
+| &nbsp;&nbsp;&nbsp;&nbsp;serialNumber | str | Yes |  |  | serial number of the device.<br>This field is required along with parentContainerName |
+| &nbsp;&nbsp;&nbsp;&nbsp;systemMacAddress | str | No |  |  | MAC address of the device |
+| &nbsp;&nbsp;&nbsp;&nbsp;parentContainerName | str | Yes |  |  | Name of the parent container.<br>This field is required along with either serialNumber or fqdn |
```
Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]: consider inline highlighting references to other variables, for example `fqdn`

File: ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py (hunk 1 above)
Comment (JulioPDX) [inline ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py]:
```suggestion
    description: List of devices with their container, configlet, and image bundle information
```

File: ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py [hunk 11]
```diff
@@ -45,7 +45,7 @@
     choices: ['present', 'factory_reset', 'provisioning_reset', 'absent']
     type: str
   apply_mode:
-    description: Set how configlets are attached/detached on device. If set to strict all configlets not listed in your vars are detached.
+    description: Set how configlets are attached/detached on device. If set to strict all configlets and image bundles not listed in your vars are detached.
```
Comment (JulioPDX) [inline ansible_collections/arista/cvp/plugins/modules/cv_device_v3.py]:
```suggestion
    description: Set how configlets are attached/detached on device. If set to strict, all configlets and image bundles not listed in your vars are detached.
```

File: ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
+# Schema for cv_change_control_v3
+
+| Variable | Type | Required | Default | Choices | Description |
+| -------- | ---- | -------- | ------- | ------------------ | ----------- |
+| change_id | list | No |  |  | List of change IDs to get/remove |
+| name | str | No |  |  | The name of the change control. If not provided, one will be generated |
```
Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]: Mixing sentence and non sentence structure here. Probably not a big deal but thought you should be aware.

File: ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md (hunk 5 above)
Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]: This list example and the ones below seem to be missing spacing. For example `-name` vs `- name`

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md (hunk 3 above)
Comment (sugetha24) [reply ansible_collections/arista/cvp/docs/schema/cv_device_v3.md:5]: No. This is a md file that I created. It's not from the docstrings

File: ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;parent | str | Yes |  |  | Name of parent stage |
+| &nbsp;&nbsp;&nbsp;&nbsp;activities | Dict | Yes |  |  |  |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-name | str | Yes |  |  | Only used internally, "task" for any tasks |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;stage | str | Yes |  |  | The name of the Stage to assign the task to |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;task_id | str | Yes |  |  | The WorkOrderId of the task to be executed, if this is to be a task activity |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;timeout | int | No | 900 |  | The timeout, if this is to be a task activity |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;action | str | Yes |  |  | The name of the action performed (mutually exclusive to task_id and timeout) |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;arguments | Dict | Yes |  |  |  |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-name | str | Yes |  |  | Device ID |
```
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]:
```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- name | str | Yes |  |  | Device ID |
```

File: ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+| state | str | No | show | show<br>set<br>remove | Set if we should get, set/update, or remove the change control |
+| change | Dict | No |  |  | A dict containing the change control to be created/modified |
+| &nbsp;&nbsp;&nbsp;&nbsp;name | str | No |  |  | Name of change control |
+| &nbsp;&nbsp;&nbsp;&nbsp;notes | str | No |  |  | Any notes that you want to add |
+| &nbsp;&nbsp;&nbsp;&nbsp;stages | Dict | Yes |  |  |  |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-name | str | Yes |  |  | Name of stage |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mode | str | Yes |  | series<br>parallel | Serial or parallel execution |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;parent | str | Yes |  |  | Name of parent stage |
+| &nbsp;&nbsp;&nbsp;&nbsp;activities | Dict | Yes |  |  |  |
+| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-name | str | Yes |  |  | Only used internally, "task" for any tasks |
```
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]:
```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- name | str | Yes |  |  | Only used internally, "task" for any tasks |
```

File: ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md (hunk 5 above)
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]:
```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- name | str | Yes |  |  | Name of stage |
```

File: ansible_collections/arista/cvp/docs/schema/cv_container_v3.md (hunk 2 above)
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_container_v3.md]:
```suggestion
| apply_mode | str | No | loose | loose<br>strict | Set how configlets are attached/detached to containers. If set to strict, all configlets not listed in your vars will be detached. |
```

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md (hunk 4 above)
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]:
```suggestion
| state | str| No | present | present<br>factory_reset<br>provisioning_reset<br>absent | Set if Ansible should build, remove devices from provisioning, fully decommission or factory reset devices on CloudVision |
```

File: mkdocs.yml [hunk 15]
```diff
@@ -137,6 +137,16 @@ nav:
...
+      - Module cv_device_v3: docs/schema/cv_device_v3.md
+      # - Module cv_task_v3: docs/schema/cv_task_v3.md
+      # - Module cv_facts_v3: docs/schema/cv_facts_v3.md
+      # - Module cv_image_v3: docs/schema/cv_image_v3.md
+      - Module cv_tag_v3: docs/schema/cv_tag_v3.md
+      - Module cv_change_control_v3: docs/schema/cv_change_control_v3.md
```
Comment (sugetha24) [inline mkdocs.yml]:
```suggestion
  - Schema documentation:
    - CV Modules v3:
      # - cv_configlet_v3: docs/schema/cv_tag_v3.md
      - cv_container_v3: docs/schema/cv_container_v3.md
      - cv_device_v3: docs/schema/cv_device_v3.md
      # - cv_task_v3: docs/schema/cv_task_v3.md
      # - cv_facts_v3: docs/schema/cv_facts_v3.md
      # - cv_image_v3: docs/schema/cv_image_v3.md
      - cv_tag_v3: docs/schema/cv_tag_v3.md
      - cv_change_control_v3: docs/schema/cv_change_control_v3.md
```

File: ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md (hunk 12 above)
Comment (sugetha24) [reply ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]: Feel free to make suggestions.

Comment (JulioPDX) [inline ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]:
```suggestion
| name | str | No |  |  | The name of the change control, if not provided, one will be generated automatically |
```

Comment (noredistribution) [reply ansible_collections/arista/cvp/docs/schema/cv_change_control_v3.md]: should we also say `generated automatically`?

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
+# Schema for cv_device_v3
+
+| Variable | Type | Required | Default | Choices | Description |
+| -------- | ---- | -------- | ------- | ------------------ | ----------- |
+| apply_mode | str | No | loose | loose<br>strict | Set how configlets are attached/detached on device. If set to strict all configlets not listed in your vars are detached |
+| search_key | str | No | hostname | fqdn<br>hostname<br>serialNumber | Key name to use to look for device in Cloudvision |
+| state | str| No | present | present<br>factory_reset | Set if ansible should build or remove devices on Cloudvision |
+| devices | List | Yes |  |  | List of devices with their container and configlets information |
+| &nbsp;&nbsp;&nbsp;&nbsp;ipAddress | str | Yes |  |  | IP address of the device |
+| &nbsp;&nbsp;&nbsp;&nbsp;fqdn | str | Yes |  |  | Fully Qualified Domain Name of the device.<br>This field is required along with parentContainerName |
```
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]:
```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;fqdn | str | Yes |  |  | Fully Qualified Domain Name of the device.<br>This field is required along with `parentContainerName` |
```

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+| &nbsp;&nbsp;&nbsp;&nbsp;serialNumber | str | Yes |  |  | serial number of the device.<br>This field is required along with parentContainerName |
```
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]:
```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;serialNumber | str | Yes |  |  | serial number of the device.<br>This field is required along with `parentContainerName` |
```

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md (hunk 10 above)
Comment (sugetha24) [inline ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]:
```suggestion
| &nbsp;&nbsp;&nbsp;&nbsp;parentContainerName | str | Yes |  |  | Name of the parent container.<br>This field is required along with either `serialNumber` or `fqdn` |
```

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md (hunk 4 above)
Comment (sugetha24) [reply ansible_collections/arista/cvp/docs/schema/cv_device_v3.md]: Added this change in another suggested change. Resolving this conversation

File: ansible_collections/arista/cvp/docs/schema/cv_device_v3.md (hunk 3 above)
Comment (noredistribution) [reply ansible_collections/arista/cvp/docs/schema/cv_device_v3.md:5]: sorry, I guess we'll need to follow the 'no sentence' style then here too? 
`Set how configlets are attached/detached to devices; if set to strict, all configlets and image bundles not listed in your vars will be detached` like that? @JulioPDX

**Review by JulioPDX — APPROVED** [review]
LGTM, thank you for all the updates

**Review by noredistribution — APPROVED** [review]
LGTM!

**Pull request merged**